release/models/platform/openconfig-platform-healthz-fault.yang

```diff
@@ -0,0 +1,318 @@
+  namespace "http://openconfig.net/yang/platform/healthz-fault";
+
+  prefix "oc-platform-healthz-fault";
+
+  import openconfig-extensions { prefix oc-ext; }
+  import openconfig-types { prefix oc-types; }
+  import openconfig-platform { prefix oc-platform; }
+  import openconfig-platform-healthz { prefix oc-platform-healthz; }
+
+  // meta
+  organization "OpenConfig working group";
+
+  contact
+    "OpenConfig working group
+    www.openconfig.net";
+
+  description
+    "This model defines device reported fault";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2025-08-05" {
+    description
+      "Initial fault schema";
+    reference "0.1.0";
+  }
+
+  identity SYMPTOM_BASE {
+    description
+      "Base identity for fault symptoms";
+  }
+
+  identity SYMPTOM_OVER_THRESHOLD {
+    base SYMPTOM_BASE;
+    description
+      "Indicates that a sensor exceeds its upper operating threshold.";
+  }
+
+  identity SYMPTOM_UNDER_THRESHOLD {
+    base SYMPTOM_BASE;
+    description
+      "Indicates that a sensor is measuring below its lower operating
+      threshold.";
+  }
+
+  identity SYMPTOM_MEMORY_ERRORS {
+    base SYMPTOM_BASE;
+    description
+      "Memory errors";
+  }
+
+  identity SYMPTOM_MISSING_COMPONENT {
+    base SYMPTOM_BASE;
+    description
+      "Component is missing";
+  }
+
+  identity SYMPTOM_MISWIRED {
+    base SYMPTOM_BASE;
+    description
+      "Component is miswired";
+  }
+
+  identity SYMPTOM_BUS_ERROR {
+    base SYMPTOM_BASE;
+    description
+      "Communication or bus error occurred on the component";
+  }
+
+  identity SYMPTOM_UNKNOWN {
+    base SYMPTOM_BASE;
+    description
+      "An unknown symptom";
+  }
+
+  identity ACTION_BASE {
+    description
+      "Base identity for fault symptoms";
+  }
+
+  identity ACTION_RESEAT {
+    base ACTION_BASE;
+    description
+      "Reseat component.";
+  }
+
+  identity ACTION_WARM_REBOOT {
+    base ACTION_BASE;
+    description
+      "Reboot that does not interrupt device operation.";
+  }
+
+  identity ACTION_COLD_REBOOT {
+    base ACTION_BASE;
+    description
+      "Reboot that does interrupt device operation.";
+  }
+
+  identity ACTION_POWER_CYCLE {
+    base ACTION_BASE;
+    description
+      "The process of turning a component off completely and then turning it back
+      on.";
+  }
+
+  identity ACTION_FACTORY_RESET {
+    base ACTION_BASE;
+    description
+      "Restores a component to its original factory settings.";
+  }
+
+  identity ACTION_REPLACE {
+    base ACTION_BASE;
+    description
+      "Replace component.";
+  }
+
+  identity OPENCONFIG_FAULT_SEVERITY {
+    description
+      "Base identity for fault severity.";
+  }
+
+  identity URGENT {
+    base OPENCONFIG_FAULT_SEVERITY;
+    description
+      "Indicates that a fault requires urgent remediation.";
+  }
+
+  identity NON_URGENT {
+    base OPENCONFIG_FAULT_SEVERITY;
+    description
+      "Indicates that a fault does not require urgent remediation.";
+  }
```

Comment:
My thinking is that it is very likely that the network management system will need to determine this according to the an operator's policies for running their network.  So my bias is this should not be included in the model.

If there are some operational scenarios that could be  quantified / measured that can be used to know the urgency then maybe these are justified.  I think we'd need to see some examples and see if there is some pattern that can be measured and modeled.